### PR TITLE
nixos/nextdns: add package override option

### PR DESCRIPTION
**Pull request by pnelson**
###### Motivation for this change

Allow the service to run other package versions more easily.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/networking/nextdns.nix [hunk 1]
```diff
@@ -12,6 +12,13 @@ in {
         default = false;
         description = "Whether to enable the NextDNS DNS/53 to DoH Proxy service.";
       };
+      package = mkOption {
+        default = pkgs.nextdns;
+        defaultText = "pkgs.nextdns";
+        example = "pkgs.nextdns";
```
Comment (aanderse) [inline nixos/modules/services/networking/nextdns.nix:18]: This isn't a good example - it is the default. Please provide an example that's different than the default.

Comment (aanderse) [reply nixos/modules/services/networking/nextdns.nix]: This new example makes me think what you're really looking for is overlays, not a `package` option :thinking:

**Comment by pnelson** [comment]
For those that stumble across this because their NextDNS version is out of date, please follow this minimal example adapted from https://gist.github.com/LnL7/e645b9075933417e7fd8f93207787581.

```nix
{ config, pkgs, ... }:

let
  # sudo nix-channel --add http://nixos.org/channels/nixpkgs-unstable nixpkgs-unstable
  # sudo nix-channel --update nixpkgs-unstable
  unstable = import <nixpkgs-unstable> {};
in {
  nixpkgs.overlays = [ (self: super: {
    nextdns = unstable.nextdns;
  }) ];
}
```

Maybe something like this would be useful to add to the [wiki](https://nixos.wiki/wiki/Overlays). This seems like a common use case that isn't covered.

**Comment by aanderse** [comment]
@pnelson adding a `package` option is fine. I just try to point out that overlays achieve the same thing. In **my** opinion the `package` option adds more value when there are multiple variants or versions.

**Comment by pnelson** [comment]
Thanks! That's not the case here. We can revisit this if multiple variants arise. Overlays are probably the better option.

**Pull request closed without merging**